### PR TITLE
fix: address a minor issue that might cause duplicate entries in cluster decisions (status)

### PR DESCRIPTION
**Pull request by michaelawyu**
### Description of your changes

This PR fixes a minor issue where duplicate entries might occur in cluster decisions when a cluster is picked and then unpicked without having a scheduling policy change.

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

- [x] Unit tests
- [x] Integration tests

### CODE REVIEW

File: pkg/scheduler/framework/frameworkutils.go [hunk 1]
```diff
@@ -296,6 +308,11 @@ func newSchedulingDecisionsFromBindings(
 			break
 		}
 
+		if _, ok := seenClusters[sc.Cluster.Name]; ok {
```
Comment (ryanzhang-oss) [inline pkg/scheduler/framework/frameworkutils.go:311]:
```suggestion
		if seenClusters[sc.Cluster.Name] {
```

File: pkg/scheduler/framework/frameworkutils.go [hunk 2]
```diff
@@ -274,12 +274,24 @@ func newSchedulingDecisionsFromBindings(
...
+	// Track clusters that have been added to the decision list.
+	//
+	// This is to avoid duplicate entries in the decision list, caused by a rare occurrence
+	// where a cluster is picked before, but later loses its status as a matching (or most
+	// preferable) cluster (e.g., a label/cluster property change) under the evaluation
+	// of **the same** scheduling policy.
+	//
+	// Note that for such occurrences, we will honor the previously made decisions in an attempt
+	// to reduce fluctations.
+	seenClusters := map[string]bool{}
```
Comment (ryanzhang-oss) [inline pkg/scheduler/framework/frameworkutils.go:286]: why not use make(map[string]bool)?

File: pkg/scheduler/framework/frameworkutils.go (hunk 1 above)
Comment (michaelawyu) [reply pkg/scheduler/framework/frameworkutils.go]: Done 👌

File: pkg/scheduler/framework/frameworkutils.go (hunk 2 above)
Comment (michaelawyu) [reply pkg/scheduler/framework/frameworkutils.go]: Done 👌

**Review by ryanzhang-oss — APPROVED** [review]

**Pull request merged**